libusb/os/linux_usbfs.c

```diff
@@ -449,21 +486,162 @@ static void op_exit(struct libusb_context *ctx)
 static int op_set_option(struct libusb_context *ctx, enum libusb_option option, va_list ap)
 {
 	UNUSED(ctx);
-	UNUSED(ap);
 
 #ifdef __ANDROID__
 	if (option == LIBUSB_OPTION_WEAK_AUTHORITY) {
-		usbi_dbg("set libusb has weak authority");
-		weak_authority = 1;
+		int parameter = va_arg(ap, int);
```

Comment:
it used to be valid using `LIBUSB_OPTION_WEAK_AUTHORITY` without any parameters like `libusb_set_option(nullptr, LIBUSB_OPTION_WEAK_AUTHORITY)` and now we're expecting at least one (int) - is it still going to be valid for old code that doesn't pass any?

Reply:
The examples I found passed a trailing parameter of `NULL`, which evaluates to zero.  If not everybody is passing the trailing `NULL` then this extra parameter should be removed.  Is there existing code as you describe?

Reply:
~~We can't know for sure - there could be private repositories that are using it.~~
~~I use a general rule for myself: if something is strictly valid, there will be someone who is using it.~~

A quick googling:
1) Perfectly valid usage:
https://github.com/IntelRealSense/RealSenseID/blob/4096914d860117ec2b3ab78fa2380ace5fae13e0/src/Capture/AndroidCapture.cc#L37
2) One of the examples you mentioned:
https://github.com/brainflow-dev/brainflow/blob/311375178ca2e88a9005ffea412f3f6cc4b0de84/src/utils/libftdi_serial.cpp#L264
3) An intuitive example, and personally I though to do exactly this:
https://github.com/IntelRealSense/RealSenseID/blob/4096914d860117ec2b3ab78fa2380ace5fae13e0/src/Capture/AndroidCapture.cc#L37

I'd prefer 3) to be correct one, but I guess to make it properly it is better to have a separate option, and probably out-of-scope of this PR.

Reply:
The change I made to WEAK_AUTHORITY's arguments should be removed.

Everything we've discussed has been added to the todo list at the top, and the other items we haven't have been marked optional.

I'm unfortunately not using this PR myself any more, but if others need me to fix it up for merge so they can use it, I can probably eventually find some time to do so.

Reply:
> Everything we've discussed has been added to the todo list at the top

This is as much as I'd expect.

I was considering using change in my project, but due to other reasons, I have to stick with wrap_sys_device.
But while my memory is fresh, I'm trying to mark here all the issues I find, just so the community (or maybe even myself, but later) will be able to finish it.

Thanks for being around!